core/templates/dev/head/domain/collection/ReadOnlyCollectionBackendApiService.ts

```diff
@@ -40,16 +44,27 @@ oppia.factory('ReadOnlyCollectionBackendApiService', [
 
       $http.get(collectionDataUrl).then(function(response) {
         var collection = angular.copy(response.data.collection);
+        _cacheCollectionDetails(response.data);
         if (successCallback) {
           successCallback(collection);
         }
+        $rootScope.$broadcast('collectionLoaded');
       }, function(errorResponse) {
         if (errorCallback) {
           errorCallback(errorResponse.data);
         }
       });
     };
 
+    var _cacheCollectionDetails = function(details) {
+      _collectionDetailsCache[details.collection.id] = {
+        canEdit: details.can_edit,
+        isLoggedIn: details.is_logged_in,
```

Comment:
I'm not sure about having `canEdit, isLoggedIn` as a part of collection data. For the first one (can_edit) we have CollectionRightsBackendAPIServices and for the later, I think we have UserInfoService (not sure!)

Reply:
I agree with this -- it's not a "collection detail". It's more of a user-related detail, and the code should reflect that.

Reply:
@seanlip, @DubeySandeep the problem with using the UserService here is that it sends a whole new request just for the logged-in status and that will delay the page load.

Reply:
Actually, I guess canEdit is a collection detail, since it is collection-specific. 

But isLoggedIn clearly is not. Perhaps you can still get that value in the relevant call but not store it as "collection details". It's a value that only needs to be read once.